pkg/common/k8s/types.go

```diff
@@ -0,0 +1,591 @@
+/*
+Copyright 2025 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package k8s
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/nuclio/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	autosv2 "k8s.io/api/autoscaling/v2"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+// ClientWithRetry provides resilient methods to interact with various Kubernetes resources
+type ClientWithRetry interface {
```

Comment:
The interface is just `Client`. The retry is the specific implementation

Reply:
Also - isn't this needed for all Nuclio CRDs as well?

Reply:
@TomerShor I replaced all the usages of kubernetes.Interface. For CRDs we use https://github.com/nuclio/nuclio/blob/37096462b8b3a1faa2d8e5a84a41e35de55b6c20/pkg/platform/kube/client/clientset/versioned/clientset.go#L31-L34, which will be handled in a separate PR